test/test.js

```diff
@@ -650,6 +651,85 @@ const createTestSuite = (useElectronNet) => {
       })
     })
 
+    it('should handle aborts before request', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      abort.abort()
+      url = `${base}timeout`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return expect(fetch(url, opts)).to.eventually.be.rejected
+        .and.be.an.instanceOf(FetchError)
+        .and.have.property('type', 'abort')
+    })
+
+    it('should handle aborts during a request', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      setTimeout(() => {
+        abort.abort()
+      }, 100)
+      url = `${base}timeout`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return expect(fetch(url, opts)).to.eventually.be.rejected
+        .and.be.an.instanceOf(FetchError)
+        .and.have.property('type', 'abort')
+    })
+
+    it('should handle aborts during a response', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      setTimeout(() => {
+        abort.abort()
+      }, 100)
+      url = `${base}slow`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return fetch(url, opts).then(res => {
+        expect(res.ok).to.be.true
+        return expect(res.text()).to.eventually.be.rejectedWith(FetchError)
+          .and.to.satisfy(e => e.message.endsWith('request aborted'))
+      })
+    })
+
+    it('should handle aborts after request finish', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      url = `${base}hello`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+
+      return fetch(url, opts).then(r => r.text()).then(r => {
+        abort.abort()
+      })
+    })
+
+    it('should handle aborts after request error', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      url = `${base}error/reset`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+
+      return expect(fetch(url, opts)).to.eventually.be.rejected
+        .and.be.an.instanceOf(FetchError)
+        .and.have.property('code', 'ECONNRESET')
+        .then(() => {
+          abort.abort()
```

Comment:
All tests should also have an assertion that there is no memory leak, at the end : `assert.deepEqual(signal.listeners['abort'], [])`

Reply:
This does not seem possible with the current implementation of AbortController